### DISCUSSION TITLE
Error installing 2.5.8 with Arduino CLI - can't find dependencies

### DISCUSSION THREAD

**Discussion by stonehippo** [SpenceKonde/megaTinyCore#623] [Library discussion]
HI @SpenceKonde, I use the Arduino CLI in addition to the IDE. I realize `arduino-cli` may not be a common target, but I wanted to let you know about an issue that I've seen trying to update to 2.5.

Normally, to update a core in Arduino CLI, you'd do something like this:

```
arduino-cli update
arduino-cli upgrade
```

These commands pull the latest board manager updates, then applies them. Pretty simple. With 2.5.8, the following error is coming up:

```
Downloading megaTinyCore:megaavr@2.5.8...
Error upgrading: Can't find dependencies for platform megaTinyCore:megaavr@2.5.8: getting tool dependencies for platform megaTinyCore:megaavr@2.5.8: tool version 6.3.0-arduino17or18 not found
```

I've run into this on both MacOS and Raspberry Pi OS (Linux/ARM). I worked around it on MacOS by installing the update via Arduino IDE.

Addendum: As I was writing this, I noticed that after install 2.5.8 via the Arduino IDE, it's showing what looks like a related error:

```
Index error: could not find referenced tool name=avrdude version=6.3.0-arduino17or18 packager=arduino
```

**Comment by SpenceKonde** [comment]
OH god fucking damnit
I will do a second emergency release, I screwed up the bloody json file.

**Reply by stonehippo** [reply]
Thanks! And thanks for all of the work you put into this core…it's an awesome resource.

**Comment by SpenceKonde** [comment]
Let's see if 2.5.9 is viable... that's 9 emergency releases since 2.5.0

**Comment by SpenceKonde** [comment]
Fix is in board manager.

**Reply by stonehippo** [reply]
2.5.9 seems to work on Raspberry Pi OS and MacOS. On the latter, I found that it wouldn't update with `arduino-cli` (in fact, I could even uninstall it with the command line). I worked around it by running the update in the IDE, so 🤷🏾.

Thanks for jumping on this one, and for all the hard work you do.

**Comment by stonehippo** [comment]
Hi @SpenceKonde, seems like the Arduino IDE is still throwing this error:

```
Index error: could not find referenced tool name=avrdude version=6.3.0-arduino17or18 packager=arduino
```

I think this is because it's still trying to parse the bad 2.5.8 entry. When you have a chance, no rush, since this isn't a functional issue, but could you update the index file with the correct packager for that entry? I don't think this needs a new release at all, just an edit to index file.

**Comment by SpenceKonde** [comment]
See my releasescripts repo issue#37, we are trying to come up with a
solution that minimizes breakage.

____________
Spence Konde
Azzy’S Electronics

New products! Check them out at tindie.com/stores/DrAzzy
GitHub: github.com/SpenceKonde
ATTinyCore: Arduino support for almost every ATTiny microcontroller
Contact: ***@***.***
